service/matching/pri_backlog_manager.go

```diff
@@ -120,11 +120,21 @@ func (c *priBacklogManagerImpl) Stop() {
 	// initialized. Also skip if we're stopping due to lost ownership (the update will
 	// fail in that case). Ignore any errors. Don't bother with GC, the next reload will
 	// handle that.
-	if c.initializedError.Ready() && !c.skipFinalUpdate.Load() {
-		ctx, cancel := context.WithTimeout(c.tqCtx, ioTimeout)
-		_ = c.db.SyncState(ctx)
-		cancel()
+	if !c.initializedError.Ready() || c.skipFinalUpdate.Load() {
+		return
 	}
+
+	c.subqueueLock.Lock()
+	for i, r := range c.subqueues {
+		_, ackLevel := r.getLevels()
+		// oldestTime can be time.Time{} here since countDelta is 0
```

Comment:
nit: the comment is a little worrying since the behavior of `updateAckLevelAndBacklogStats` could change later. I'd sleep better knowing that the default time value would always be ignored regardless of countDelta.

Reply:
True, but I didn't want to add a whole new function just to update ackLevel alone. Would you prefer it that way?

Reply:
Isn't that also part of `updateAckLevelAndBacklogStats` behavior that could change later? As the caller the only way to be confident seems to be another function that doesn't even have those parameters